src/System.Private.CoreLib/shared/System/Collections/Generic/List.cs

```diff
@@ -554,14 +560,13 @@ public void ForEach(Action<T> action)
             }
 
             int version = _version;
-
-            for (int i = 0; i < _size; i++)
+            foreach (T item in new ReadOnlySpan<T>(_items, 0, _size)) // Using Span to avoid additional bounds checks from using List
```

Comment:
This will likely regress performance for small sizes.

Reply:
@benaadams, can you share perf numbers?

Reply:
(Or undo the controversial changes to get in the rest.)

Reply:
> (Or undo the controversial changes to get in the rest.)

Backed out the `List<T>` changes as was mostly an exercise in using `CollectionsMarshal` and that didn't anymore.